core/src/main/java/io/confluent/kafka/schemaregistry/storage/LookupStore.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2018 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.confluent.kafka.schemaregistry.storage;
+
+import io.confluent.kafka.schemaregistry.client.rest.entities.Schema;
+
+import java.util.List;
+
+/**
+ * Internal interface that provides various indexed methods that help lookup the underlying schemas.
```

Comment:
it looks like this interface does a lot more than lookup underlying schemas -- there are callbacks for deletions, etc. maybe we should update the javadoc to reflect it?

Reply:
thinking more about this, would it be better to call this interface a cache? it's implementations don't actually persist schemas, right? and the contract is that it can be used to keep schemas in memory which were originally pulled out of the persistent store (Kafka).

Reply:
LookupCache?

Reply:
Either sound good. Just a minor nit: SchemaRegistryCache sounds like a public interface, whereas here that's probably not the intent.